app/javascript/dashboard/store/modules/conversations/index.js

```diff
@@ -202,6 +214,19 @@ export const mutations = {
   [types.CLEAR_CONVERSATION_FILTERS](_state) {
     _state.appliedFilters = [];
   },
+
+  [types.SET_LAST_MESSAGE_ID_IN_SYNC_CONVERSATION](
+    _state,
+    { conversationId, messageId }
+  ) {
+    _state.syncConversationsMessages[conversationId] = messageId;
+  },
+  [types.CLEAR_LAST_MESSAGE_ID_FROM_SYNC_CONVERSATION](
+    _state,
+    { conversationId }
+  ) {
+    _state.syncConversationsMessages[conversationId] = null;
+  },
```

Comment:
Do you know if we need to maintain this separately? Can we not just infer it from the `allConversations` state?

Reply:
Yes. We need to maintain this separation. The reconnect logic is not reliable at the moment. It would be better to store the last message id at the disconnect time so that it can fetch all the messages after the last id when reconnect happens.